src/main/java/com/fasterxml/jackson/databind/deser/std/NumberDeserializers.java

```diff
@@ -655,18 +661,54 @@ protected final Float _parseFloat(JsonParser p, DeserializationContext ctxt)
     @JacksonStdImpl
     public static class DoubleDeserializer
         extends PrimitiveOrWrapperDeserializer<Double>
+        implements ContextualDeserializer
     {
         private static final long serialVersionUID = 1L;
+        protected final JsonFormat.Value _jsonFormat;
+        protected final DecimalFormat _decimalFormat;
 
         final static DoubleDeserializer primitiveInstance = new DoubleDeserializer(Double.TYPE, 0.d);
         final static DoubleDeserializer wrapperInstance = new DoubleDeserializer(Double.class, null);
-        
+
         public DoubleDeserializer(Class<Double> cls, Double nvl) {
             super(cls, LogicalType.Float, nvl, 0.d);
+            _jsonFormat = null;
+            _decimalFormat = null;
+        }
+
+        protected DoubleDeserializer(Class<Double> cls, Double nvl, JsonFormat.Value format) {
+            super(cls, LogicalType.Float, nvl, 0.d);
+            _jsonFormat = format;
+
+            Locale locale = _jsonFormat.getLocale();
+            if (locale == null) {
+                locale = Locale.getDefault();
+            }
+            _decimalFormat = new DecimalFormat("", new DecimalFormatSymbols(locale));
+        }
+
+        @Override
+        public JsonDeserializer<?> createContextual(DeserializationContext ctxt, BeanProperty property)
+                throws JsonMappingException
+        {
+            final JsonFormat.Value format = findFormatOverrides(ctxt, property, handledType());
+            if (format != null) {
+                if (_primitive) {
+                    return new DoubleDeserializer(Double.TYPE, 0.d, format);
+                }
+                return new DoubleDeserializer(Double.class, null, format);
+            }
+            return this;
         }
 
         @Override
         public Double deserialize(JsonParser p, DeserializationContext ctxt) throws IOException {
+            if (_jsonFormat != null) {
+                Double result = _parseDoubleFormat(p, ctxt);
+                if (result != null) {
+                    return result;
+                }
+            }
             if (p.hasToken(JsonToken.VALUE_NUMBER_FLOAT)) {
```

Comment:
The order here needs to be swapped around: start with the fast, expected case and only check for formatting if 
JSON String value encountered.